lib/components-schema-v1_7_x.ts

```diff
@@ -399,45 +441,30 @@ const componentPropertyDefinition: {
         type: 'string',
         description: 'Feature flag that should be present for the property to show up. Always show if not specified.',
     },
-};
-// Define childProperties after the definition of componentPropertyDefinition to avoid
-// a circular reference when creating a copy of componentPropertyDefinition
-// Currently childProperties is only supported one level deep.
-componentPropertyDefinition.childProperties = {
-    type: 'array',
-    description: 'List of conditional child properties',
-    items: {
-        type: 'object',
-        properties: {
-            matchType: {
-                type: 'string',
-                description: `Defines how to match the parent property's value`,
-                enum: ['exact-value'],
-            },
-            matchExpression: {
-                type: ['boolean', 'integer', 'string', 'number'],
-                description: `The expression to use to match the parent property's value`,
-            },
+    childProperties: {
```

Comment:
Not really sure how this is done in the rest of the validator, but are/should there be tests to validate the schema? For example testing with a definition that contains an invalid `matchType`.

Reply:
I think we have tests in https://github.com/WoodWing/studio-component-set-tools/pull/93.

Reply:
Positive flow test is indeed in #93, this PR restricts the possibilities. The question is what negative tests would bring us. Wouldn't that be just testing whether `AJV` does its job?

Reply:
Good point.